File: myeslib3-stack1/src/test/java/myeslib3/stack1/Stack1WriteModelRepositoryIt.java
package myeslib3.stack1;

import com.google.gson.Gson;
import com.google.inject.Guice;
import com.google.inject.Injector;
import myeslib3.core.data.UnitOfWork;
import myeslib3.core.data.Version;
import myeslib3.examples.example1.core.aggregates.customer.CreateCustomerCmd;
import myeslib3.examples.example1.core.aggregates.customer.CustomerCreated;
import myeslib3.examples.example1.runtime.CustomerModule;
import myeslib3.stack1.infra.DatabaseModule;
import org.junit.Before;
import org.junit.Test;
import org.skife.jdbi.v2.DBI;

import javax.inject.Inject;
import java.util.Arrays;

public class Stack1WriteModelRepositoryIt {

	final static Injector injector = Guice.createInjector(new CustomerModule(),
					new Stack1Module(),
					new DatabaseModule());

	@Inject
	Gson gson;
	@Inject
	DBI dbi;

	Stack1WriteModelRepository repo ;

	@Before
	public void setup() {
			injector.injectMembers(this);
			repo = new Stack1WriteModelRepository("customer", gson, dbi);
	}

	@Test
	public void append() {

		String id = "c1";
		CreateCustomerCmd command = new CreateCustomerCmd("c1");
		CustomerCreated event = new CustomerCreated(id, command.getName());
		UnitOfWork uow1 = UnitOfWork.create(id, command.getName(), command, Version.create(1),  Arrays.asList(event));

		repo.append(uow1);

	}
}

File: myeslib3-stack1/src/test/kotlin/myeslib3/Stack1WriteModelRepositoryIt.kt
package myeslib3

import com.google.gson.Gson
import com.google.inject.Guice
import myeslib3.core.data.UnitOfWork
import myeslib3.core.data.Version
import myeslib3.examples.example1.core.aggregates.customer.*
import myeslib3.examples.example1.runtime.CustomerModule
import myeslib3.stack1.Stack1Module
import myeslib3.stack1.Stack1WriteModelRepository
import myeslib3.stack1.infra.CamelModule
import myeslib3.stack1.infra.DatabaseModule
import org.junit.jupiter.api.Test
import org.skife.jdbi.v2.DBI

class Stack1WriteModelRepositoryIt {

    @Test
    fun aTest() {

        val injector = Guice.createInjector(Stack1Module(), CustomerModule(), CamelModule(), DatabaseModule())

        val dbi: DBI = injector.getInstance(DBI::class.java)
        val gson = injector.getInstance(Gson::class.java)

        val repository = Stack1WriteModelRepository("customer", gson, dbi)

        repository.append(uow2())

    }

    fun uow1(): UnitOfWork {
        val cmd: CreateCustomerCmd = CreateCustomerCmd("customer1")
        val customer = dependencyInjectionFn.inject(Customer())
        val version = Version.create(0)
        return commandHandlerFn.handle(commandId, cmd, customerId, customer, version, stateTransitionFn, dependencyInjectionFn).result
    }

    fun uow2(): UnitOfWork {
        val cmd = CreateActivatedCustomerCmd("customer3", "because I also want it")
        val customer = dependencyInjectionFn.inject(Customer())
        val version = Version.create(0)
        return commandHandlerFn.handle(commandId, cmd, "customer#3", customer, version, stateTransitionFn, dependencyInjectionFn).result
    }


}
